doc/api/esm.md

```diff
@@ -1235,29 +1247,28 @@ export async function getFormat(url, context, defaultGetFormat) {
     };
   }
   // Defer to Node.js for all other URLs.
-  return defaultGetFormat(url, context, defaultGetFormat);
+  return null;
 }
 ```
 
-#### <code>getSource</code> hook
+#### <code>getSource(url, context)</code>
 
 > Note: The loaders API is being redesigned. This hook may disappear or its
 > signature may change. Do not rely on the API described below.
 
+* `url` {string}
+* `context` {Object}
+  * `format` {string}
+* Returns: {Object} If `null` is returned, Node.js will defer to the parent
+  loader.
+  * `source` {string|Buffer}
```

Comment:
This seems to imply that the loader needs to potentially load everything twice - once to determine the format and then to load the source. Can we make `getFormat` optional and let the `getSource` hook return the format as well?

Reply:
Yeah I don't think getFormat is fantastic in its current state. I am planning to look into that in the future, not here.